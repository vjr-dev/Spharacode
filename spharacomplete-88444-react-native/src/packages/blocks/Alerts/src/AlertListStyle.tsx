// Customizable Area Start
//@ts-ignore
//@ts-nocheck
import { Dimensions, StyleSheet } from "react-native";
import { RFValue } from "react-native-responsive-fontsize";
import { COLORS } from "../../../framework/src/Globals";
const sh = Dimensions.get("window").height;
const sw = Dimensions.get("window").width;

export const Styles = StyleSheet.create({
  container: {
    height: sh,
    width: sw,
    flex: 1,
  },
  safeareaView: {
    flex: 1,
    backgroundColor: "#f17234",
  },
  modalView: {
    position: "absolute",
    bottom: 15,
    width: sw - 30,
    alignSelf: "center",
  },
  menuView: {
    width: "100%",
    borderRadius: 15,
    backgroundColor: COLORS.Viewback,
    justifyContent: "center",
    alignItems: "center",
  },
  menuText: {
    padding: 15,
    fontSize: RFValue(15),
    color: COLORS.white,
  },
  seperator: {
    height: 1,
    backgroundColor: COLORS.skipGray,
    width: "90%",
  },
  cancelButton: {
    justifyContent: "center",
    alignItems: "center",
    width: "100%",
    backgroundColor: "#fed99f",
    borderRadius: 25,
  },
  cancelText: {
    paddingVertical: 17,
    fontSize: RFValue(13),
  },
  shadowView: {
    backgroundColor: "#3e3e3e",
    width: sw,
    height: sh,
    position: "absolute",
    opacity: 0.6,
    // zIndex:1,
    flex: 1,
  },
  header: {
    marginLeft: RFValue(10),
    fontSize: RFValue(14),
    color: COLORS.infoGray,
  },
});
// Customizable Area End
