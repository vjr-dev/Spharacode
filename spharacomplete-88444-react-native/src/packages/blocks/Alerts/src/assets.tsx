// Customizable Area Start
export const imgPasswordVisible = require("../assets/ic_password_visible.png");
export const imgPasswordInVisible = require("../assets/ic_password_invisible.png");
export const user = require("../assets/user.png");
export const back1 = require("../assets/image_background-1.png");
export const back2 = require("../assets/image_background-2.png");
export const image_back = require("../assets/image_back.png");
export const alert_icon = require("../assets/alert_icon.png");
// Customizable Area End