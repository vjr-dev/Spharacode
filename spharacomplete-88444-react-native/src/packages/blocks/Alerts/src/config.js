// Customizable Area Start
Object.defineProperty(exports, "__esModule", {
  value: true,
});
exports.urlGetAlertHistory ="bx_block_dashboard/dashboards/emergency_assistance_alerts";
exports.GetAlertHistoryApiContentType = "application/json";
exports.GetAlertHistoryApiMethodType = "GET";
// Customizable Area End
