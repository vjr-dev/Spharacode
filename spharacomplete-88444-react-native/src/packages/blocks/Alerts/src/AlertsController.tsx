// Customizable Area Start
//@ts-ignore
//@ts-nocheck
import { IBlock } from "../../../framework/src/IBlock";
import { BlockComponent } from "../../../framework/src/BlockComponent";
import MessageEnum, {
  getName,
} from "../../../framework/src/Messages/MessageEnum";
import { runEngine } from "../../../framework/src/RunEngine";

export const configJSON = require("./config");

export interface Props {
  navigation: any;
  id: string;
}

interface S {
  txtInputValue: string;
  txtSavedValue: string;
  enableField: boolean;
  Detail: boolean;
  Chat: boolean;
}

interface SS {
  id: any;
}

export default class AlarmController extends BlockComponent<Props, S, SS> {
  constructor(props: Props) {
    super(props);
    this.receive = this.receive.bind(this);
    this.subScribedMessages = [getName(MessageEnum.AccoutLoginSuccess)];

    this.state = {
      txtInputValue: "",
      txtSavedValue: "A",
      enableField: false,
      Detail: true,
      Chat: false,
    };
    runEngine.attachBuildingBlock(this as IBlock, this.subScribedMessages);
  }
}
// Customizable Area End
