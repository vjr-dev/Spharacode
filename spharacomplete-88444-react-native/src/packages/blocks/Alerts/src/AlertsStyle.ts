// Customizable Area Start
//@ts-ignore
//@ts-nocheck
import { Dimensions, StyleSheet } from "react-native";
import { RFPercentage, RFValue } from "react-native-responsive-fontsize";
import { COLORS } from "../../../framework/src/Globals";
const sh = Dimensions.get("window").height;
const sw = Dimensions.get("window").width;

export const Styles = StyleSheet.create({
  container: {
    height: sh,
    width: sw,
    flex: 1,
  },
  header_view: {
    height: 60,
    width: sw,
    justifyContent: "space-between",
    alignItems: "center",
    backgroundColor: COLORS.darkorange,
    flexDirection: "row",
  },
  header_txt: {
    alignSelf: "center",
    color: COLORS.ReviewBlack,
    fontWeight: "500",
    fontSize: RFValue(14),
    marginRight: RFPercentage(19),
  },
  top_tab: {
    flexDirection: "row",
    backgroundColor: COLORS.backgroundGray,
    height: sh * 0.04,
    justifyContent: "space-around",
    alignItems: "center",
    marginTop: RFPercentage(1),
  },
  public_txt: {
    color: COLORS.skipGray,
    fontSize: 17,
    marginBottom: 7,
    alignSelf: "center",
    fontWeight: "600",
  },
  private_txt: {
    color: COLORS.lightyellow,
    fontSize: 17,
    marginBottom: 7,
    alignSelf: "center",
    fontWeight: "600",
  },
  private_view: {
    borderBottomColor: COLORS.lightyellow,
    borderBottomWidth: 2,
    width: sw * 0.18,
  },
  public_view: {
    borderBottomColor: COLORS.lightyellow,
    borderBottomWidth: 2,
    width: sw * 0.18,
  },
  alarm_view: {
    marginTop: RFPercentage(3.5),
    marginLeft: RFPercentage(2.5),
  },
  alram_txt: {
    color: COLORS.white,
    fontSize: RFValue(15),
    fontWeight: "600",
  },
  details_main_view: {
    marginTop: RFPercentage(2),
    marginLeft: RFPercentage(2.1),
    justifyContent: "space-between",
  },
  date_view: {
    flexDirection: "row",
    justifyContent: "space-between",
    width: "90%",
  },
  val_txt: {
    marginTop: RFValue(10),
    color: COLORS.infoGray,
  },
  info_txt: {
    marginRight: RFPercentage(20),
    marginTop: RFValue(10),
    color: COLORS.infoGray,
    alignSelf: "flex-end",
  },
  info_txt1: {
    marginRight: RFPercentage(0),
    marginTop: RFValue(10),
    color: COLORS.infoGray,
    alignSelf: "flex-end",
  },
  info_txt2: {
    marginRight: RFPercentage(10),
    marginTop: RFValue(10),
    color: COLORS.infoGray,
  },
  main_view: {
    borderWidth: 3,
    borderColor: COLORS.darkorange,
    borderRadius: 20,
    height: RFPercentage(19),
    width: sw * 0.87,
    alignSelf: "center",
    marginTop: RFPercentage(4),
    backgroundColor: COLORS.lightorangeback,
  },
  alert_view: {
    justifyContent: "center",
    alignItems: "center",
    marginTop: RFPercentage(1),
  },
  alert_txt: {
    color: COLORS.skipGray,
    fontSize: RFValue(13),
    fontWeight: "500",
    paddingHorizontal: 10,
    textAlign: "center",
  },
});
// Customizable Area End
