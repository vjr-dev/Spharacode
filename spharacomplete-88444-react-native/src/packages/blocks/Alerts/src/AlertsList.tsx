// Customizable Area Start
//@ts-ignore
//@ts-nocheck
import React from "react";
import {
  ActivityIndicator,
  Dimensions,
  Image,
  ImageBackground,
  Modal,
  SafeAreaView,
  ScrollView,
  Text,
  TouchableOpacity,
  View,
} from "react-native";

import { RFValue } from "react-native-responsive-fontsize";
import { heightPercentageToDP as hp } from "react-native-responsive-screen";
import { COLORS } from "../../../framework/src/Globals";
import { Styles } from "./AlertListStyle";
import AlertsListController, { Props } from "./AlertsListController";
import * as IMAGE from "./assets";
const windowWidth = Dimensions.get("window").width;
const windowHeight = Dimensions.get("window").height;
const sw = Dimensions.get("window").width;

export default class AlertList extends AlertsListController {
  constructor(props: Props) {
    super(props);
  }
  renderAlerts(item: any, index: number) {
    return (
      <View
        key={it.id}
        style={{
          marginBottom: 10,
          marginTop: 5,
          paddingBottom: 10,
          paddingTop: 10,
          backgroundColor: "#3F4040",
          justifyContent: "center",
        }}
      >
        <TouchableOpacity
          testID="onPressAlert"
          onPress={() =>
            this.props.navigation.navigate("Alerts", {
              date: it.date,
              time: it.time,
              name: it.status_report,
              address: it.address,
              alertType: it.alert_type,
            })
          }
        >
          <View
            style={{
              marginHorizontal: 12,
              flexDirection: "column",
            }}
          >
            <Text
              style={{
                fontSize: 15,
                fontWeight: "600",
                color: COLORS.white,
              }}
            >
              {it.date}
            </Text>
            <View
              style={{
                width: "100%",
                flexDirection: "row",
              }}
            >
              <Text
                style={{
                  marginTop: 5,
                  flex: 1,
                  fontSize: 13,
                  fontWeight: "600",
                  color: COLORS.infoGray,
                }}
              >
                {it.address}
              </Text>
            </View>
          </View>
        </TouchableOpacity>
      </View>
    );
  }
  render() {
    return (
      <SafeAreaView style={Styles.safeareaView}>
        <ImageBackground source={IMAGE.back1} style={Styles.container}>
          <ImageBackground source={IMAGE.back2} style={Styles.container}>
            <View
              style={{
                height: hp("7%"),
                width: sw,
                justifyContent: "space-between",
                flexDirection: "row",
                backgroundColor: COLORS.darkorange,
                alignItems: "center",
              }}
            >
              <TouchableOpacity
                testID="alertsListBackButton"
                style={{
                  alignItems: "center",
                  height: "90%",
                  justifyContent: "center",
                  width: "15%",
                }}
                onPress={() => this.props.navigation.pop()}
              >
                <Image
                  source={IMAGE.image_back}
                  style={{
                    height: 15,
                    width: 15,
                    tintColor: COLORS.backgroundGray,
                  }}
                />
              </TouchableOpacity>
              <View
                style={{
                  height: "100%",
                  width: "70%",
                  alignItems: "center",
                  justifyContent: "center",
                }}
              >
                <Text
                  style={{
                    color: COLORS.backgroundGray,
                    fontWeight: "700",
                    fontSize: RFValue(16),
                  }}
                >
                  My Alert
                </Text>
              </View>
              <View style={{ height: "100%", width: "15%" }} />
            </View>

            {this.state.Loader ? (
              <View style={{ flex: 1 }}>
                <Modal visible={this.state.Loader} transparent={true}>
                  <View
                    style={{
                      height: windowHeight,
                      width: windowWidth,
                      alignItems: "center",
                      justifyContent: "center",
                    }}
                  >
                    <View
                      style={{
                        height: 100,
                        width: windowWidth - 20,
                        alignSelf: "center",
                        justifyContent: "center",
                        alignItems: "center",
                      }}
                    >
                      <ActivityIndicator
                        animating={true}
                        size={"large"}
                        color="#f07233"
                      />
                    </View>
                  </View>
                </Modal>
              </View>
            ) : (
              <ScrollView
                style={{
                  width: "100%",
                  marginTop: 15,
                  alignSelf: "center",
                  flex: 1,
                }}
                showsVerticalScrollIndicator={false}
                bounces={false}
              >
                {[Object.keys(this.state.Channels)].map(
                  (item: any, index: number) => {
                    return (
                      <View key={index.toString() + "_outer_id"}>
                        <Text style={Styles.header}>{item}</Text>
                        {this.state.Channels[item]?.map(
                          (it: any, index: number) => {
                            return this.renderAlerts(item, index);
                          }
                        )}
                      </View>
                    );
                  }
                )}
                {/* <SectionList
                  sections={this.state.Channels}
                  keyExtractor={(item, index) => item + index}
                  renderItem={renderItem}
                  renderSectionHeader={({ section: { date } }) => (
                    <Text style={Styles.header}>{date}</Text>
                    // <Text style={Styles.header}>{Obje}</Text>
                  )}
                /> */}
              </ScrollView>
            )}
          </ImageBackground>
        </ImageBackground>
      </SafeAreaView>
    );
  }
}
// Customizable Area End
