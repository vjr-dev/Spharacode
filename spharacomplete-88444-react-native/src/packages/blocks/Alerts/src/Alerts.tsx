// Customizable Area Start
//@ts-ignore
//@ts-nocheck
import React from "react";
import {
  Dimensions,
  Image,
  ImageBackground,
  SafeAreaView,
  ScrollView,
  Text,
  TouchableOpacity,
  View,
} from "react-native";

import { RFPercentage, RFValue } from "react-native-responsive-fontsize";
import { heightPercentageToDP as hp } from "react-native-responsive-screen";
import { COLORS } from "../../../framework/src/Globals";
import AlertsController, { Props } from "./AlertsController";
import { Styles } from "./AlertsStyle";
import * as IMAGE from "./assets";
const sh = Dimensions.get("window").height;
const sw = Dimensions.get("window").width;

export default class Alerts extends AlertsController {
  constructor(props: Props) {
    super(props);
  }

  render() {
    return (
      <ImageBackground source={IMAGE.back1} style={Styles.container}>
        <ImageBackground source={IMAGE.back2} style={Styles.container}>
          <SafeAreaView>
            <ScrollView style={{ marginBottom: RFPercentage(1), height: sh }}>
              {/* <View
                style={Styles.header_view}
              >
                <TouchableOpacity style={{ marginLeft: 0, height: 60, width: 60, alignItems: "center", justifyContent: "center" }} onPress={() => this.props.navigation.pop()}>
                  <Icon name="chevron-left" color={COLORS.black} size={18} />
                </TouchableOpacity>
                <Text style={Styles.header_txt}>11-08-2020</Text>
              </View> */}
              <View
                style={{
                  height: hp("7%"),
                  width: sw,
                  justifyContent: "space-between",
                  flexDirection: "row",
                  backgroundColor: COLORS.darkorange,
                  alignItems: "center",
                }}
              >
                <TouchableOpacity
                  testID="alertsBackButton"
                  style={{
                    alignItems: "center",
                    height: "90%",
                    justifyContent: "center",
                    width: "15%",
                  }}
                  onPress={() => this.props.navigation.pop()}
                >
                  <Image
                    source={IMAGE.image_back}
                    style={{
                      height: 15,
                      width: 15,
                      tintColor: COLORS.backgroundGray,
                    }}
                  />
                </TouchableOpacity>
                <View
                  style={{
                    height: "100%",
                    width: "70%",
                    alignItems: "center",
                    justifyContent: "center",
                  }}
                >
                  <Text
                    style={{
                      color: COLORS.backgroundGray,
                      fontWeight: "700",
                      fontSize: RFValue(16),
                    }}
                  >
                    {this.props.route.params?.date}
                  </Text>
                </View>
                <View style={{ height: "100%", width: "15%" }} />
              </View>
              <View style={Styles.top_tab}>
                <TouchableOpacity
                  testID="alertsDetailsButton"
                  onPress={() => this.setState({ Detail: true, Chat: false })}
                  style={[
                    Styles.public_view,
                    {
                      borderBottomColor: this.state.Detail
                        ? COLORS.lightyellow
                        : COLORS.backgroundGray,
                    },
                  ]}
                >
                  <Text
                    style={[
                      Styles.public_txt,
                      {
                        color: this.state.Detail
                          ? COLORS.lightyellow
                          : COLORS.skipGray,
                      },
                    ]}
                  >
                    Detail
                  </Text>
                </TouchableOpacity>
                <TouchableOpacity
                  style={[
                    Styles.private_view,
                    {
                      borderBottomColor: this.state.Chat
                        ? COLORS.lightyellow
                        : COLORS.backgroundGray,
                    },
                  ]}
                >
                  <Text
                    style={[
                      Styles.private_txt,
                      {
                        color: this.state.Chat
                          ? COLORS.lightyellow
                          : COLORS.skipGray,
                      },
                    ]}
                  >
                    Chat
                  </Text>
                </TouchableOpacity>
              </View>
              {this.state.Detail ? (
                <View style={{ width: "100%" }}>
                  <View style={Styles.alarm_view}>
                    <Text style={Styles.alram_txt}>Alert detail</Text>
                  </View>
                  <View style={Styles.details_main_view}>
                    <View style={Styles.date_view}>
                      <Text style={{ color: COLORS.infoGray }}>Date :</Text>
                      <Text
                        style={{
                          marginRight: RFPercentage(15),
                          color: COLORS.infoGray,
                          alignSelf: "flex-end",
                        }}
                      >
                        {this.props.route.params?.date}
                      </Text>
                    </View>
                    <View style={Styles.date_view}>
                      <Text style={Styles.val_txt}>Time :</Text>
                      <Text style={Styles.info_txt}>
                        {this.props.route.params?.time}
                      </Text>
                    </View>
                    <View style={Styles.date_view}>
                      <Text style={Styles.val_txt}>Emergency called # : </Text>
                      <Text
                        style={[
                          Styles.info_txt1,
                          { marginRight: RFPercentage(20.5) },
                        ]}
                      >
                        {this.props.route.params?.alertType}
                      </Text>
                    </View>
                    <View style={Styles.date_view}>
                      <Text
                        style={[
                          Styles.val_txt,
                          { marginRight: RFPercentage(8) },
                        ]}
                      >
                        Location :
                      </Text>
                      <Text style={[Styles.info_txt2, {}]}>
                        {this.props.route.params?.address}
                      </Text>
                    </View>
                    <View style={Styles.date_view}>
                      <Text style={Styles.val_txt}>Status report:</Text>
                      <Text
                        style={[
                          Styles.info_txt,
                          { marginRight: RFPercentage(21.5) },
                        ]}
                      >
                        {this.props.route.params?.name}
                      </Text>
                    </View>
                  </View>
                </View>
              ) : (
                <View>
                  <View style={Styles.main_view}>
                    {/* <Icon name="exclamation-triangle" color={COLORS.orange} size={RFValue(31)} style={{ alignSelf: 'center', marginTop: RFValue(10) }} /> */}
                    <Image
                      source={IMAGE.alert_icon}
                      style={{
                        alignSelf: "center",
                        marginTop: RFValue(10),
                        height: RFValue(30),
                        width: RFValue(30),
                      }}
                    ></Image>
                    <Text
                      style={{
                        alignSelf: "center",
                        fontSize: RFValue(14),
                        fontWeight: "500",
                        marginTop: RFValue(4),
                      }}
                    >
                      New Alert!
                    </Text>
                    <View style={Styles.alert_view}>
                      <Text style={Styles.alert_txt}>
                        SPHARA!!! I'm being followed. I'm scared. Send help!
                      </Text>
                      {/* <Text style={Styles.alert_txt}>scared. Send help!</Text> */}
                    </View>
                  </View>
                </View>
              )}
            </ScrollView>
          </SafeAreaView>
        </ImageBackground>
      </ImageBackground>
    );
  }
}
// Customizable Area End
