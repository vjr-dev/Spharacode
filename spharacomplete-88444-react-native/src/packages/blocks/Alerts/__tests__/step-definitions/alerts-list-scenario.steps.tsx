//@ts-ignore
//@ts-nocheck
import { shallow, ShallowWrapper } from "enzyme";
import { defineFeature, loadFeature } from "jest-cucumber";
import * as helpers from "../../../../framework/src/Helpers";
import { Message } from "../../../../framework/src/Message";
import { runEngine } from "../../../../framework/src/RunEngine";
import React from "react";
import { View } from "react-native";
import MessageEnum, {
  getName,
} from "../../../../framework/src/Messages/MessageEnum";
import AlertsList from "../../src/AlertsList";
const alertData = {
  all_alerts: [
    {
      panic_incidents: [
        {
          id: 3877,
          date: "07-February-23",
          time: "10:48AM",
          alert_type: "PanicIncident",
          address: "Varanasi, Sadar, Varanasi, Uttar Pradesh, 221010, India",
          status_report: "Pending",
        },
        {
          id: 3876,
          date: "07-February-23",
          time: "10:47AM",
          alert_type: "PanicIncident",
          address: "Varanasi, Sadar, Varanasi, Uttar Pradesh, 221010, India",
          status_report: "Pending",
        },
        {
          id: 3875,
          date: "07-February-23",
          time: "10:44AM",
          alert_type: "PanicIncident",
          address: "Varanasi, Sadar, Varanasi, Uttar Pradesh, 221010, India",
          status_report: "Pending",
        },
        {
          id: 3867,
          date: "30-January-23",
          time: "10:44AM",
          alert_type: "PanicIncident",
          address: "Varanasi, Sadar, Varanasi, Uttar Pradesh, 221010, India",
          status_report: "Pending",
        },
        {
          id: 3862,
          date: "27-January-23",
          time: "01:41PM",
          alert_type: "PanicIncident",
          address: "Varanasi, Sadar, Varanasi, Uttar Pradesh, 221010, India",
          status_report: "Pending",
        },
        {
          id: 3827,
          date: "25-January-23",
          time: "04:14PM",
          alert_type: "PanicIncident",
          address: "Varanasi, Sadar, Varanasi, Uttar Pradesh, 221010, India",
          status_report: "Pending",
        },
        {
          id: 3826,
          date: "25-January-23",
          time: "04:13PM",
          alert_type: "PanicIncident",
          address: "Varanasi, Sadar, Varanasi, Uttar Pradesh, 221010, India",
          status_report: "Pending",
        },
        {
          id: 3825,
          date: "25-January-23",
          time: "04:08PM",
          alert_type: "PanicIncident",
          address: "Varanasi, Sadar, Varanasi, Uttar Pradesh, 221010, India",
          status_report: "Pending",
        },
        {
          id: 3803,
          date: "23-January-23",
          time: "11:34AM",
          alert_type: "PanicIncident",
          address: "Varanasi, Sadar, Varanasi, Uttar Pradesh, 221010, India",
          status_report: "Pending",
        },
        {
          id: 3799,
          date: "20-January-23",
          time: "04:06PM",
          alert_type: "PanicIncident",
          address:
            "My (Ashwani Kumar Ojha) House, B38/55-40      , C/O :  Shrimati Amrawati Ojha, Raja Sir Motichand Road, Varanasi, Sadar, Varanasi, Uttar Pradesh, 221010, India",
          status_report: "Pending",
        },
        {
          id: 3797,
          date: "20-January-23",
          time: "03:38PM",
          alert_type: "PanicIncident",
          address: "Varanasi, Sadar, Varanasi, Uttar Pradesh, 221010, India",
          status_report: "Pending",
        },
      ],
    },
  ],
  status: 200,
};

const screenProps = {
  navigation: {
    dispatch: jest.fn(),
    getParam: jest.fn(),
    navigate: jest.fn(),
    goBack: jest.fn(),
    pop:jest.fn(),
    addListener: jest.fn().mockImplementation((event: any, callback: any) => {
      callback();
    }),
  },
  id: "Alerts",
};

const feature = loadFeature(
  "./__tests__/features/alerts-list-scenario.feature"
);

defineFeature(feature, (test) => {
  beforeEach(() => {
    jest.resetModules();
    jest.doMock("react-native", () => ({ Platform: { OS: "ios" } }));
    jest.spyOn(helpers, "getOS").mockImplementation(() => "android");
  });

  test("User navigates to Alerts List Screen", ({ given, when, then }) => {
    let alertsWrapper: ShallowWrapper;
    let instance: AlertsList;

    given("I am a User getting alerts list", () => {
      alertsWrapper = shallow(<AlertsList {...screenProps} />);
      expect(alertsWrapper).toBeTruthy();
      expect(alertsWrapper).toMatchSnapshot();
    });

    when("I navigate to the Alerts List Screen", () => {
      instance = alertsWrapper.instance() as AlertsList;
      instance.setState({
        Channels: alertData.all_alerts,
      });
      expect(alertsWrapper).toBeTruthy();
      expect(alertsWrapper).toMatchSnapshot();
    });

    then("I can press back button", () => {
      let alertsListBackButton = alertsWrapper.findWhere(
        (node) => node.prop("testID") === "alertsListBackButton"
      );
      alertsListBackButton.simulate("press");
      expect(alertsListBackButton).toBeTruthy();
    });

    then("Get alert history with out errors", () => {
      const getAlertHistoryAPI: Message = new Message(
        getName(MessageEnum.RestAPIResponceMessage)
      );
      getAlertHistoryAPI.addData(
        getName(MessageEnum.RestAPIResponceDataMessage),
        getAlertHistoryAPI.messageId
      );
      getAlertHistoryAPI.addData(
        getName(MessageEnum.RestAPIResponceSuccessMessage),
        JSON.parse(JSON.stringify({}))
      );
      instance.MessageId = getAlertHistoryAPI.messageId;
      runEngine.sendMessage("Unit Test", getAlertHistoryAPI);
      instance.receiveGetAlertHistory(alertData);
    });

    then("Get alert history with errors", () => {
      const getAlertHistoryAPI: Message = new Message(
        getName(MessageEnum.RestAPIResponceMessage)
      );
      getAlertHistoryAPI.addData(
        getName(MessageEnum.RestAPIResponceDataMessage),
        getAlertHistoryAPI.messageId
      );
      getAlertHistoryAPI.addData(
        getName(MessageEnum.RestAPIResponceErrorMessage),
        JSON.parse(JSON.stringify({}))
      );
      instance.MessageId = getAlertHistoryAPI.messageId;
      runEngine.sendMessage("Unit Test", getAlertHistoryAPI);
      instance.receiveGetAlertHistory(null);
    });

    then("I can press on alert", () => {
      let listWrapper = shallow(
        <View>
          {instance.renderAlerts(
            {
              id: 3877,
              date: "07-February-23",
              time: "10:48AM",
              alert_type: "PanicIncident",
              address:
                "Varanasi, Sadar, Varanasi, Uttar Pradesh, 221010, India",
              status_report: "Pending",
            },
            0
          )}
        </View>
      );
      let onPressAlert = listWrapper.findWhere(
        (node) => node.prop("testID") === "onPressAlert"
      );
      onPressAlert.simulate("press");
      expect(onPressAlert).toBeTruthy();
    });
  });
});
