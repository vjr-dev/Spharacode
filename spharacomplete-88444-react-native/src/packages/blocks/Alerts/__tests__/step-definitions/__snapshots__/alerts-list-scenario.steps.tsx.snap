// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alerts List User navigates to Alerts List Screen 1`] = `
<SafeAreaView
  style={
    Object {
      "backgroundColor": "#f17234",
      "flex": 1,
    }
  }
>
  <ImageBackground
    source={
      Object {
        "testUri": "../../../packages/blocks/Alerts/assets/image_background-1.png",
      }
    }
    style={
      Object {
        "flex": 1,
        "height": 1334,
        "width": 750,
      }
    }
  >
    <ImageBackground
      source={
        Object {
          "testUri": "../../../packages/blocks/Alerts/assets/image_background-2.png",
        }
      }
      style={
        Object {
          "flex": 1,
          "height": 1334,
          "width": 750,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "rgba(255,138,73,1)",
            "flexDirection": "row",
            "height": undefined,
            "justifyContent": "space-between",
            "width": 750,
          }
        }
      >
        <TouchableOpacity
          activeOpacity={0.2}
          onPress={[Function]}
          style={
            Object {
              "alignItems": "center",
              "height": "90%",
              "justifyContent": "center",
              "width": "15%",
            }
          }
          testID="alertsListBackButton"
        >
          <Image
            source={
              Object {
                "testUri": "../../../packages/blocks/Alerts/assets/image_back.png",
              }
            }
            style={
              Object {
                "height": 15,
                "tintColor": "#404040",
                "width": 15,
              }
            }
          />
        </TouchableOpacity>
        <View
          style={
            Object {
              "alignItems": "center",
              "height": "100%",
              "justifyContent": "center",
              "width": "70%",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#404040",
                "fontSize": undefined,
                "fontWeight": "700",
              }
            }
          >
            My Alert
          </Text>
        </View>
        <View
          style={
            Object {
              "height": "100%",
              "width": "15%",
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <Component
          hardwareAccelerated={false}
          transparent={true}
          visible={true}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "height": 1334,
                "justifyContent": "center",
                "width": 750,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "height": 100,
                  "justifyContent": "center",
                  "width": 730,
                }
              }
            >
              <ActivityIndicator
                animating={true}
                color="#f07233"
                hidesWhenStopped={true}
                size="large"
              />
            </View>
          </View>
        </Component>
      </View>
    </ImageBackground>
  </ImageBackground>
</SafeAreaView>
`;

exports[`Alerts List User navigates to Alerts List Screen 2`] = `
<SafeAreaView
  style={
    Object {
      "backgroundColor": "#f17234",
      "flex": 1,
    }
  }
>
  <ImageBackground
    source={
      Object {
        "testUri": "../../../packages/blocks/Alerts/assets/image_background-1.png",
      }
    }
    style={
      Object {
        "flex": 1,
        "height": 1334,
        "width": 750,
      }
    }
  >
    <ImageBackground
      source={
        Object {
          "testUri": "../../../packages/blocks/Alerts/assets/image_background-2.png",
        }
      }
      style={
        Object {
          "flex": 1,
          "height": 1334,
          "width": 750,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "rgba(255,138,73,1)",
            "flexDirection": "row",
            "height": undefined,
            "justifyContent": "space-between",
            "width": 750,
          }
        }
      >
        <TouchableOpacity
          activeOpacity={0.2}
          onPress={[Function]}
          style={
            Object {
              "alignItems": "center",
              "height": "90%",
              "justifyContent": "center",
              "width": "15%",
            }
          }
          testID="alertsListBackButton"
        >
          <Image
            source={
              Object {
                "testUri": "../../../packages/blocks/Alerts/assets/image_back.png",
              }
            }
            style={
              Object {
                "height": 15,
                "tintColor": "#404040",
                "width": 15,
              }
            }
          />
        </TouchableOpacity>
        <View
          style={
            Object {
              "alignItems": "center",
              "height": "100%",
              "justifyContent": "center",
              "width": "70%",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#404040",
                "fontSize": undefined,
                "fontWeight": "700",
              }
            }
          >
            My Alert
          </Text>
        </View>
        <View
          style={
            Object {
              "height": "100%",
              "width": "15%",
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <Component
          hardwareAccelerated={false}
          transparent={true}
          visible={true}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "height": 1334,
                "justifyContent": "center",
                "width": 750,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "height": 100,
                  "justifyContent": "center",
                  "width": 730,
                }
              }
            >
              <ActivityIndicator
                animating={true}
                color="#f07233"
                hidesWhenStopped={true}
                size="large"
              />
            </View>
          </View>
        </Component>
      </View>
    </ImageBackground>
  </ImageBackground>
</SafeAreaView>
`;
